docs/dev/building-on-zksync/contracts/differences-with-ethereum.md

```diff
@@ -97,29 +98,29 @@ export function createAddress(sender: Address, senderNonce: BigNumberish) {
 
 ### Differences in `block.timestamp` and `block.number`
 
-These variables behave differently from L1. Read more [about blocks in zkSync](../../developer-guides/transactions/blocks.md#blocks-in-zksync-era).
+You can find more information about blocks in zkSync, including the differences in block.timestamp and block.number, [in this link.](../../developer-guides/transactions/blocks.md#blocks-in-zksync-era).
 
 ### Differences in `COINBASE` `DIFFICULTY` `BASEFEE` 
+In zkSync, there are some differences in the behavior of certain variables compared to L1.
 
-- `COINBASE` returns our Bootloader contract address, `0x0000000000000000000000000000000000008001` 
-- `DIFFICULTY` returns a constant value of `2500000000000000` 
-- `BASEFEE` is not a constant, but is defined by the fee model. Most of the time it is 0.25 gwei, but under very high L1 gas prices it may rise.
+- The `COINBASE` variable in zkSync returns the address of the Bootloader contract, which is `0x0000000000000000000000000000000000008001`.
 
-### Use of `codesize`/`codecopy` and `calldata` in the deploy code
+- The `DIFFICULTY` variable returns a constant value of `2500000000000000`.
 
-`Calldata` in the constructor is not empty as on the EVM, on zkEVM calldata contains constructor arguments.
-`codesize` has the same behavior as `calldatasize`, `codecopy` as `calldatacopy`.
-To make it works datasize of the current object is 0.
+- The `BASEFEE` variable is not a constant in zkSync and is instead defined by the fee model. Usually, it is set to 0.25 gwei, but it can increase under very high L1 gas prices.
 
-### Using `return` in the deploy code will not function as intended
+### Use of `codesize`/`codecopy` and `calldata` in the deploy code
+On zkEVM, the behavior of `calldata` and `codesize`/`codecopy` instructions in the constructor is different compared to EVM. In zkEVM, `calldata` contains constructor arguments, while in EVM, `calldata` is empty during constructor execution.
 
-Constructors on the zkSync Era return immutables arrays. If you attempt to use return, it will instead return the array of immutables in the auxiliary heap that was previously written there (using `setimmutable`) rather than the data you specified.
+In addition, `codesize` instruction behaves similarly to `calldatasize`, and `codecopy` instruction behaves similarly to `calldatacopy`. To make this work, the datasize of the current object needs to be set to 0.
```

Comment:
```suggestion
In addition, `codesize` instructions behave similarly to `calldatasize`, and `codecopy` instructions behave similarly to `calldatacopy`. To make this work, the datasize of the current object needs to be set to 0.
```

Reply:
What's the **this** that you are making work? Please clarify.

Reply:
Done!